### ISSUE TITLE
Ubuntu 15.04 (Vivid) fails to install

### ISSUE THREAD

**Issue by justonian** [dnschneid/crouton#1706]
Hi there-

I realize I'm an early adopter of vivid, but I've tried twice with different parameters and can't get 15.04 with unity to install via crouton. Detailed notes regarding my requested install and error output is as follows. Appreciate any guidance on how to unblock the setup.

My command to invoke the chroot setup: sudo sh -e ~/Downloads/crouton -r vivid -t unity,keyboard,touch,xiwi -n vivid

Setup progresses as expected during retrieve/validate/extracting phases (outside of a keyring file not available warning presented a few times) until it gets to unpacking and configuring xkb-data where it mentions the python3 package may be at fault. Message there presented below:

I: Unpacking xkb-data...
W: Failure while installing base packages.  This will be re-attempted up to five times.
W: See //debootstrap/debootstrap.log for details (possibly the package /var/cache/apt/archives/python3_3.4.3-1_amd64.deb is at fault)

Next, it continues on and eventually fails again at ureadahead. Message show below:
I: Configuring ureadahead...
W: Failure while configuring base packages.  This will be re-attempted up to five times.
W: See //debootstrap/debootstrap.log for details (possibly the package /var/cache/apt/archives/python3_3.4.3-1_amd64.deb is at fault)

Finally, all GIDs get updated successfully then the final critical failure is relating to lsb-release, which ultimately fails installation and fails the chroot setup overall.

dpkg: dependency problems prevent configuration of lsb-release:
 lsb-release depends on python3; however:
  Package python3 is not installed.
 lsb-release depends on python3:any (>= 3.3.2-2~); however:

dpkg: error processing package lsb-release (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of ubuntu-minimal:
 ubuntu-minimal depends on lsb-release; however:
  Package lsb-release is not configured yet.
 ubuntu-minimal depends on python3; however:
  Package python3 is not installed.
 ubuntu-minimal depends on resolvconf; however:
  Package resolvconf is not installed.

dpkg: error processing package ubuntu-minimal (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of dh-python:
 dh-python depends on python3:any (>= 3.3.2-2~).

dpkg: error processing package dh-python (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 lsb-release
 ubuntu-minimal
 dh-python
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/vivid...

**Comment by dnschneid** [comment]
Yeah, vivid's fundamentally broken at the moment.  Something appears to be wrong with python3...could you find and post the output of  the file `/debootstrap/debootstrap.log` inside your chroot?  You should be able to browse it as root without actually entering the chroot.

**Comment by 1rocketdude** [comment]
I got the same error when trying to install vivid - couldn't find debootstrap.log anywhere on my system

See //debootstrap/debootstrap.log for details (possibly the package /var/cache/apt/archives/python3_3.4.3-1_amd64.deb is at fault

**Comment by Justonian** [comment]
I searched within the chroot (both under the /mnt/stateful_partition/crouton/chroots/ and /usr/local/chroots) and could not find any references to debootstrap.

Any guidance on where this might be found?

**Comment by dnschneid** [comment]
Try editing the chroot's /prepare.sh to drop into a shell after a failing `debootstrap --second-stage` so you can explore the temp filesystem that it creates.

**Comment by dortzur** [comment]
+1

**Comment by dragonloverlord** [comment]
I have managed to get a working vivid chroot (image attached) by upgrading a trusty chroot to utopic then to vivid.
![screenshot 2015-05-04 at 10 56 10 am](https://cloud.githubusercontent.com/assets/2529736/7455710/7b757db2-f24c-11e4-941c-8a8560a5568b.png)

**Comment by dragonloverlord** [comment]
@dnschneid I think this is an issue with crouton as vivid works if you just update a trusty chroot to utopic then vivid.

**Comment by dnschneid** [comment]
Actually, I think it's an issue with debootstrap.

**Comment by dortzur** [comment]
If I understand correctly this release marks the transition to systemd,
which will finally allow us to run services normally. I would hate to wait
for the next LTS for that.

**Comment by dnschneid** [comment]
Technically correct.  Once crouton supports systemd, running services the normal way should be trivial.

Unfortunately, switching to systemd breaks everything and would require crouton to make drastic changes, like #364.  Totally independent of whether systemd is a good thing or a bad thing (please don't discuss that here), it has to be launched as pid 1 and thus can't be used in a simple chroot environment on a non-systemd system.

So yes, once crouton transitions to proper systemd support, service launching will be trivial.  But given the limited development resources we have, and the high quality and usability of the LTS trusty release, if there's some hacky way to make most desktop environments launch in Vivid, that will probably be the limit of support.  This is also why utopic is still not a fully-supported release.

**Comment by raphael** [comment]
I'm able to get passed that issue by running `apt-get -f install` in the chroot, the following shows the relevant commands:
```
sudo enter-chroot -n vivid -u root
Would you like to finish the setup? [Y/n/d] n
root@localhost:~# apt-get -f install
After this operation, 370 kB of additional disk space will be used.
Do you want to continue? [Y/n] y 
WARNING: The following packages cannot be authenticated!
  python3 resolvconf
Install these packages without verification? [y/N] y
root@localhost:~# exit
chronos@localhost / $ sudo enter-chroot -n vivid
Would you like to finish the setup? [Y/n/d] y
Preparing chroot environment...
Installing target core...
Preparing environment...
...
```

**Comment by raphael** [comment]
I also had to tweak the `gnome` target to use the package `gnome-flashback` instead of `gnome-fallback`.

**Comment by dnschneid** [comment]
That makes me wonder if debootstrap isn't pulling in a package with extra Ubuntu keys.  There may be a way to specify extra packages for debootstrap to pull in if that's the case.

The latter issue is #1689 (which you probably already know, but for other people's reference)

**Comment by dnschneid** [comment]
@dortzur: meanwhile, there's [this, apparently](https://github.com/smaknsk/servicectl)

**Comment by dortzur** [comment]
Thanks! Will try it. Thanks for the amazing works so far.

**Comment by vrcod** [comment]
+1

**Comment by tonyxue** [comment]
Just take a look at this issue for the first time. I had this already but I just forgot about that.

>  it has to be launched as pid 1

So I guess 3.8 would make it possible?
But well, if that's the case, we don't even need chroot anymore, am I correct?

**Comment by dnschneid** [comment]
Yeah, the stuff needed to get a new PID space would effectively replace the call to chroot as well.

**Comment by masterdome** [comment]
@raphael How do you tweak the gnome target to use the package gnome-flashback instead of gnome-fallback.

**Comment by tonyxue** [comment]
@masterdome 
See https://github.com/dnschneid/crouton/pull/1689#discussion_r29560691
There's a syntax you can use.

**Comment by masterdome** [comment]
@tonyxue Like i dont know how to do it :(

**Comment by raphael** [comment]
@tonyxue: you have to enter the chroot and edit the file `/prepare.sh` as indicated in the ticket:
Once the build fails do:
```
sudo enter-chroot -n vivid -u root
nano /prepare.sh # search for gnome-session-fallback and replace with gnome-session-flashback
```
(this assumes you are using the default chroot name, if not replace `vivid` with the name you chose).

**Comment by masterdome** [comment]
@raphael It says -su: nano: command not found

**Comment by raphael** [comment]
try installing it `sudo apt-get install nano`.

**Comment by masterdome** [comment]
@raphael  Then what do i put?

**Comment by raphael** [comment]
I'm not sure this issue is the right place for this type of hand holding. You should have everything you need from all the answers above. Edit the file and replace fallback with flashback then exit the chroot and resume the install. As a side note installing crouton may not be the best place to get started with Linux :)

**Comment by masterdome** [comment]
@raphael K thanks

**Comment by tonyxue** [comment]
@raphael 
> you have to enter the chroot and edit the file /prepare.sh as indicated in the ticket

Yes, if you have already setup the chroot.
The link I sent is talking about tweaking the target itself.

**Comment by raphael** [comment]
ah yes that's even better, thanks for the clarification.

**Comment by MarkGavalda** [comment]
Nowadays it gets stuck at:
```
Package xserver-xorg-video-modesetting is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  xserver-xorg-core:i386 xserver-xorg-core

E: Package 'xserver-xorg-video-modesetting' has no installation candidate
Failed to complete chroot setup.
```

**Comment by gb9k** [comment]
@MarkGavalda how did you get it working? Pointing modsettings to xserver-xorg-core or commenting out the line?

**Comment by MarkGavalda** [comment]
@gb9k pointing it to xorg-core! and you'll have to follow @rafael 's guide for apt-get install -f as well.

**Comment by tista500** [comment]
@MarkGavalda Really thanks!! I've pointed it to xserver-xorg-core and it finished... :)

**Comment by MarkGavalda** [comment]
@tista500 great to hear! and does it start up properly? I found that mine gets stuck before loading Unity correctly. :( I can do some stuff but not everything (there's no navigation / start panel for example...)

**Comment by tista500** [comment]
@MarkGavalda My Asus C200 already had Utopic Gnome so, now I've got budgie-desktop (surely gnome-shell as well).

![screenshot from 2015-06-30 20-37-12](https://cloud.githubusercontent.com/assets/6073622/8430005/ec4e3086-1f67-11e5-9c0c-fd5affa128c0.png)

Regards.

**Comment by telebuss** [comment]
I am also trying to install vivid and have had no success.
I have firstly installed trusty and then done a release upgrade. After the installation, when I call startunity this is what I get:

chronos@localhost / $ sudo startunity
Entering /mnt/stateful_partition/crouton/chroots/vivid...
X.Org X Server 1.17.1
Release Date: 2015-02-10
X Protocol Version 11, Revision 0
Build Operating System: Linux 3.2.0-61-generic x86_64 Ubuntu
Current Operating System: Linux localhost 3.10.18 #1 SMP Tue Jun 30 01:51:01 PDT 2015 x86_64
Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 2506752 verity payload=PARTUUID=3795d8b8-67da-7c4f-bca7-7bf7bb4ef3b0/PARTNROFF=1 hashtree=PARTUUID=3795d8b8-67da-7c4f-bca7-7bf7bb4ef3b0/PARTNROFF=1 hashstart=2506752 alg=sha1 root_hexdigest=1beec3172d7b8dfd41877c5a99b87730bcb20716 salt=0644f88bb381a5ea2f4cc4ddee6b5660debf5fa349f8cf44000f1b49403bf17a" noinitrd vt.global_cursor_default=0 kern_guid=3795d8b8-67da-7c4f-bca7-7bf7bb4ef3b0 add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic  
Build Date: 19 March 2015  09:26:59AM
xorg-server 2:1.17.1-0ubuntu3 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.32.6
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Fri Jul 10 17:43:17 2015
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
(EE) 
(EE) Backtrace:
(EE) 0: /usr/bin/Xorg (xorg_backtrace+0x56) [0x7fc0be1fa556]
(EE) 1: /usr/bin/Xorg (0x7fc0be047000+0x1b7749) [0x7fc0be1fe749]
(EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (0x7fc0bbb0a000+0x352f0) [0x7fc0bbb3f2f0]
(EE) 3: /usr/bin/Xorg (0x7fc0be047000+0xb9879) [0x7fc0be100879]
(EE) 4: /usr/bin/Xorg (xf86BusProbe+0x9) [0x7fc0be0d3239]
(EE) 5: /usr/bin/Xorg (InitOutput+0x734) [0x7fc0be0e1d34]
(EE) 6: /usr/bin/Xorg (0x7fc0be047000+0x5c0aa) [0x7fc0be0a30aa]
(EE) 7: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf0) [0x7fc0bbb2aa40]
(EE) 8: /usr/bin/Xorg (0x7fc0be047000+0x4662e) [0x7fc0be08d62e]
(EE) 
(EE) Segmentation fault at address 0x0
(EE) 
Fatal server error:
(EE) Caught signal 11 (Segmentation fault). Server aborting
(EE) 
(EE) 
Please consult the The X.Org Foundation support 
         at http://wiki.x.org
 for help. 
(EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
(EE) 
(EE) Server terminated with error (1). Closing log file.
/usr/bin/xinit: giving up
/usr/bin/xinit: unable to connect to X server: Connection refused
/usr/bin/xinit: server error
Unmounting /mnt/stateful_partition/crouton/chroots/vivid...
Sending SIGTERM to processes under /mnt/stateful_partition/crouton/chroots/vivid...

Just to clear things out, the chroot is called vivid because I have installed today trusty.
I have also checked the log file Xorg.crouton.1.log and this is the log itself:


[ 63870.311]
X.Org X Server 1.17.1
Release Date: 2015-02-10
[ 63870.320] X Protocol Version 11, Revision 0
[ 63870.322] Build Operating System: Linux 3.2.0-61-generic x86_64 Ubuntu
[ 63870.322] Current Operating System: Linux localhost 3.10.18 #1 SMP Tue Jun 30 01:51:01 PDT 2015 x86_64
[ 63870.322] Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_veri$
[ 63870.322] Build Date: 19 March 2015  09:26:59AM
[ 63870.322] xorg-server 2:1.17.1-0ubuntu3 (For technical support please see http://www.ubuntu.com/support)
[ 63870.322] Current version of pixman: 0.32.6
[ 63870.322]    Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
[ 63870.322] Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[ 63870.323] (++) Log file: "/tmp/Xorg.crouton.1.log", Time: Fri Jul 10 17:43:17 2015
[ 63870.323] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
[ 63870.324] (==) No Layout section.  Using the first Screen section.
[ 63870.324] (==) No screen section available. Using defaults.
[ 63870.324] (**) |-->Screen "Default Screen Section" (0)
[ 63870.324] (**) |   |-->Monitor "<default monitor>"
[ 63870.324] (==) No device specified for screen "Default Screen Section".
        Using the first device section listed.
[ 63870.324] (**) |   |-->Device "Intel Graphics SNA+TearFree"
[ 63870.324] (==) No monitor specified for screen "Default Screen Section".
        Using a default monitor configuration.
[ 63870.324] (==) Automatically adding devices
[ 63870.324] (==) Automatically enabling devices
[ 63870.324] (==) Automatically adding GPU devices
[ 63870.324] (WW) The directory "/usr/share/fonts/X11/cyrillic" does not exist.
[ 63870.324]    Entry deleted from font path.
[ 63870.324] (WW) The directory "/usr/share/fonts/X11/100dpi/" does not exist.
[ 63870.324]    Entry deleted from font path.
[ 63870.324] (WW) The directory "/usr/share/fonts/X11/75dpi/" does not exist.
[ 63870.324]    Entry deleted from font path.
[ 63870.324] (WW) The directory "/usr/share/fonts/X11/100dpi" does not exist.
[ 63870.324]    Entry deleted from font path.
[ 63870.324] (WW) The directory "/usr/share/fonts/X11/75dpi" does not exist.
[ 63870.324]    Entry deleted from font path.
[ 63870.324] (==) FontPath set to:
        /usr/share/fonts/X11/misc,
        /usr/share/fonts/X11/Type1,
        built-ins
[ 63870.324] (==) ModulePath set to "/usr/lib/x86_64-linux-gnu/xorg/extra-modules,/usr/lib/xorg/extra-modules,/usr/lib/xorg/modules"
[ 63870.324] (II) The server relies on udev to provide the list of input devices.
        If no devices become available, reconfigure udev or disable AutoAddDevices.
[ 63870.324] (II) Loader magic: 0x7fc0be481d80
[ 63870.324] (II) Module ABI versions:
[ 63870.324]    X.Org ANSI C Emulation: 0.4
[ 63870.324]    X.Org Video Driver: 19.0
[ 63870.324]    X.Org XInput driver : 21.0
[ 63870.324]    X.Org Server Extension : 9.0
[ 63870.325] (II) xfree86: Adding drm device (/dev/dri/card0)
[ 63870.325] (II) xfree86: Adding drm device (/dev/dri/card1)
[ 63870.326] (EE)
[ 63870.326] (EE) Backtrace:
[ 63870.326] (EE) 0: /usr/bin/Xorg (xorg_backtrace+0x56) [0x7fc0be1fa556]
[ 63870.326] (EE) 1: /usr/bin/Xorg (0x7fc0be047000+0x1b7749) [0x7fc0be1fe749]
[ 63870.326] (EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (0x7fc0bbb0a000+0x352f0) [0x7fc0bbb3f2f0]
[ 63870.326] (EE) 3: /usr/bin/Xorg (0x7fc0be047000+0xb9879) [0x7fc0be100879]
[ 63870.326] (EE) 4: /usr/bin/Xorg (xf86BusProbe+0x9) [0x7fc0be0d3239]
[ 63870.327] (EE) 5: /usr/bin/Xorg (InitOutput+0x734) [0x7fc0be0e1d34]
[ 63870.327] (EE) 6: /usr/bin/Xorg (0x7fc0be047000+0x5c0aa) [0x7fc0be0a30aa]
[ 63870.327] (EE) 7: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf0) [0x7fc0bbb2aa40]
[ 63870.327] (EE) 8: /usr/bin/Xorg (0x7fc0be047000+0x4662e) [0x7fc0be08d62e]
[ 63870.327] (EE)
[ 63870.327] (EE) Segmentation fault at address 0x0
[ 63870.327] (EE)
Fatal server error:
[ 63870.327] (EE) Caught signal 11 (Segmentation fault). Server aborting
[ 63870.327] (EE)
[ 63870.327] (EE)
Please consult the The X.Org Foundation support
         at http://wiki.x.org
 for help.
[ 63870.327] (EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
[ 63870.327] (EE)


Sorry to write something this long, but I have no idea what it can be, nor have I found anything helpful. 
I am running on dev-channel, which might be a bit of a pain in the ass for crouton, I know.
If anyone has any ideas on how to solve it, I'll be listening!

PS: I have also tried to follow some instructions given in this same discussion, but none of them have worked for me so far.

**Comment by tista500** [comment]
@telebuss If you'd completely upgraded trusty to vivid, then you should update crouton to build some stuffs like CRAS and freon...etc.

#### 1. Get start to run update on crosh:
```
sudo sh ~/Downloads/crouton -u -n vivid
```
This would be **failed** and stopped incompletely, but it's normal. Because vivid alrealy purged `xserver-xorg-video-modesetting` package.

#### 2.  Edit prepare.sh in root directory of chroot:
As Mark said (https://github.com/dnschneid/crouton/issues/1706#issuecomment-114441462), we could finish crouton installation script by editing `prepare.sh`.

Even if crouton script stopped as incomplete, you would be inside the chroot already, so you should edit remained file named `prepare.sh` on root(`/`) directory. Then open this file with editor, and seek the word "xsever-xorg-video-modesetting" so, replace it to "xserver-xorg-core" and save.

Finally exec this script `sudo ./prepare.sh` if I remembered well.  And you're lucky, crouton installation would be finished.

Regards.

**Comment by MarkGavalda** [comment]
@tista500 that looks really nice! :) :+1:

**Comment by telebuss** [comment]
@tista500, it worked! 
I feel a bit embarrassed since the problem is just mentioned above what I wrote, but thanks a lot for kindly giving an answer.

**Comment by telebuss** [comment]
@MarkGavalda, did you manage to start unity correctly?
I think I might be having the same issue as you do.
I can only see the wallpaper, and the only thing I can do is right click, so that the menu drops down, and the items highlight, but no mouse can be seen.

**Comment by MarkGavalda** [comment]
@telebuss that's *exactly* how far I got. Not sure what the issue is though :(

**Comment by tista500** [comment]
@MarkGavalda and @telebuss

I'm now tracking unity issue that you guys filed above, so please see this diff of current gnome-session 3.14.0-2ubuntu5 on Vivid:
http://launchpadlibrarian.net/205106383/gnome-session_3.14.0-2ubuntu4_3.14.0-2ubuntu5.diff.gz

Unity devs seems to drop "compiz" entry from `/usr/share/gnome-session/sessions/ubuntu.session`:
```
[GNOME Session]
Name=Ubuntu
RequiredComponents=unity-settings-daemon;
DesktopName=Unity
```
Although this change would fix some bugs, but I suppose that we still need "compiz" as RequiredComponents. Anyway please revert like this and give it a try:
```
[GNOME Session]
Name=Ubuntu
RequiredComponents=unity-settings-daemon;compiz;
DesktopName=Unity
```

Finally I don't use Unity at all, so I can't see any details though... :tongue: 
Regards.

**Comment by telebuss** [comment]
@tista500 @MarkGavalda 
Right now I'm trying to install xfce, to at least find out whether it is only in Unity where it fails or it's in other desktops too. I'll report as soon as I installed xfce, then I will try with Unity and the idea @tista500 wrote above.
Thanks a lot!

**Comment by telebuss** [comment]
Ok, so I tried both unity and xfce
@MarkGavalda, I don't know how much you "need" unity, but xfce works. Didn't play with it to check, but at least it did work.
@tista500, adding compiz to ubuntu.session made the lateral bar visible, but nothing more. 
Thanks for the try, though.
How did you get it to work? I'd like to try it that way.

**Comment by tista500** [comment]
@telebuss Ah, I've read a bug report https://github.com/dnschneid/crouton/issues/1706, ok, my suggestion was *completely* wrong. so please remove "compiz" entry again because compiz would be kicked by upstart job(!?) instead of gnome-session.

>How did you get it to work? I'd like to try it that way.

I'm using Gnome-Shell 3.16.3 (This will be landed on Wily cycle), and NOT using Unity. Stable gnome-shell 3.14 of Vivid may run out of the box on vivid/crouton with `sudo startgnome` if you added gnome target via crouton (even though we still need to tweak such `prepare.sh` to fix gnome-session-flashback issue). Anyway gnome-shell could be handled by gnome-session perfectly without any others like upstart jobs. On the other side, talking about Unity, it would need some more tricks to run without a help of display-manager (i.e. gdm, lightdm...etc) and init processes, maybe...

If I were you, I would try this way out:
1. Run gnome-terminal from session file via `/usr/share/gnome-session/sessions/ubuntu.session`
2. Get start some trials: something like `sudo service compiz start`, `unity --replace` and so.
3. See output of `pstree` to check which is the parent process in both cases "succeed" or "failed".
4. See output of `status unity7`.
...etc.

I can't answer clearly now since I haven't any clue about Unity, but there will be much more things we can do. :smile: 

Regards.

**Comment by tista500** [comment]
@telebuss Forgot to mention. If you had installed `systemd-sysv` package, please give it a try: `sudo apt-get install upstart-sysv`. It may be affected to init process.

Regards.

**Comment by telebuss** [comment]
@tista500 Sorry for the late response, but right now I'm struggling with both final exams and heat.. :(
I tried installing upstart-sysv, but nothing happened.
Another interesting fact is that no matter what session I install (vivid, xfce, etc.), I cannot return to the main ChromeOS page. That's probably because I run on dev channel.
And here's what I got when trying unity --replace
```
(vivid)hamperfait@localhost:/usr/share/gnome-session/sessions$ sudo unity --replace
WARNING: no DISPLAY variable set, setting it to :0
stop: Unknown job: unity-panel-service
compiz (core) - Info: Loading plugin: core
compiz (core) - Info: Starting plugin: core
compiz (core) - Fatal: Couldn't open display :0
compiz (core) - Info: Stopping plugin: core
compiz (core) - Info: Unloading plugin: core
start: Unknown job: unity-panel-service
```

And when I try to call unity --verbose (with additional debug output from unity):

```
(vivid)hamperfait@localhost:/usr/share/gnome-session/sessions$ sudo unity -v
WARNING: no DISPLAY variable set, setting it to :0
stop: Unknown job: unity-panel-service
compiz (core) - Info: Loading plugin: core
compiz (core) - Debug: Trying to load core from: /home/hamperfait/.compiz-1/plugins/libcore.so
compiz (core) - Debug: dlopen failed: /home/hamperfait/.compiz-1/plugins/libcore.so: cannot open shared object file: No such file or directory
compiz (core) - Debug: Trying to load core from: /usr/lib/x86_64-linux-gnu/compiz/libcore.so
compiz (core) - Debug: dlopen failed: /usr/lib/x86_64-linux-gnu/compiz/libcore.so: cannot open shared object file: No such file or directory
compiz (core) - Debug: Trying to load core from: /usr/lib/compiz/libcore.so
compiz (core) - Debug: dlopen failed: /usr/lib/compiz/libcore.so: cannot open shared object file: No such file or directory
compiz (core) - Info: Starting plugin: core
compiz (core) - Debug: Started plugin: core
compiz (core) - Fatal: Couldn't open display :0
compiz (core) - Info: Stopping plugin: core
compiz (core) - Debug: Stopped plugin: core
compiz (core) - Info: Unloading plugin: core
start: Unknown job: unity-panel-service
```

I guess the no DISPLAY is because I am running it from the ChromeOS terminal, but I have no idea, to be sincere.

**Comment by tista500** [comment]
@telebuss No Problem :smiley: 
And thanks for your tracking! It seems to lack some libraries (like `/usr/lib/x86_64-linux-gnu/compiz/libcore.so`)? I suppose this lack would be unrelated to starting from crosh without $DISPLAY...

So please read the launchpad log: https://bugs.launchpad.net/ubuntu/+source/unity/+bug/1292396 Especially from **Comment 2**:
> I did an "apt-get install ubuntu-desktop" + delete /home/user/.config/dconf/user restart + Desktop was back. All other things didn't help.

Although that bug report was for Trusty, but it might be a valuable thing to give a try... 

Regards.

**Comment by telebuss** [comment]
@tista500 I'm afraid installing ubuntu-desktop did not work. I installed it, but still I was getting the same error about libcore0.so
Any ideas about how to fix the "black screen" after entering the chroot? It is really annoying having to restart the chromebook each time I try to enter the chroot. (although yes, I know, Chromebooks boot really fast and thank god it's not windows with a 5400 HDD, that would take forever, but still:D )

**Comment by tista500** [comment]
@telebuss Today I'm installing & trying to run Unity7 as well. 

![screenshot from 2015-07-16 20-59-54](https://cloud.githubusercontent.com/assets/6073622/8722308/23616c62-2bfe-11e5-904a-9c845e0da66c.jpg)

Yeah as a picture above, I didn't add unity target into crouton but installed chroot locally by apt-getting, then I've tweaked some files to kick unity off. To be honest, I won't do this because sometimes unity breaks my favorite gnome3 environments. :p

The point is our dconf settings I now think. we should backup and move `~/.config/dconf/user` first, and reboot, login into chroot with freon... 2nd, I don't know why, the "unity-panel-service" (`/usr/lib/unity/unity-panel-service`) failed to be loaded automatically, so I've kicked it by adding autostart file. Anyway 3D-accel seems to be fine on my BayTrail-M OpenGL driver so, we need to polish the init processes to run unity7 properly...

On seeking some init processes to run unity, the config files to be placed in `/usr/share/upstart`. I'm really surprised that how many settings unity wanted to run. And my custom `/usr/local/bin/startgnome` setting to test unity is here:
```
#!/bin/sh -e
# Copyright (c) 2015 The crouton Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# Unity
# exec crouton-noroot gnome-session-wrapper ubuntu
export DESKTOP_SESSION=ubuntu
export UBUNTU_MENUPROXY=1
export GTK_MODULES="unity-gtk-module"
# export DISPLAY=:1
exec crouton-noroot \
    xinit /etc/X11/xinit/xinitrc unity-session
```
And `unity-session` is simply:
````
#!/bin/sh
# Copyright (c) 2015 The crouton Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
#
# Provides a wrapper around ubuntu-session for test

gnome-session --session=ubuntu
```

Finally, for me, it's enough to dislike unity, and I want to purge them ASAP... :smile:

Regards.

**Comment by CoolOppo** [comment]
Has any progress been made? I can't figure out how to get Vivid or Wily to work, but they're the only distros that aren't madly outdated.

**Comment by smibarber** [comment]
We have unity on xenial now!
